src/covjson_pydantic/domain.py

```diff
@@ -93,11 +93,20 @@ def check_axis(domain_type, axes, required_axes, allowed_axes, single_value_axes
             if axis is None:
                 raise ValueError(f"A '{domain_type.value}' must have a '{axis_name}'-axis.")
             if axis_name in single_value_axes:
-                if not (isinstance(axis, ValuesAxis) and len(axis.values) == 1):
+                if isinstance(axis, ValuesAxis) and len(axis.values) != 1:
+                    raise ValueError(
+                        f"The 'values' field of the ValuesAxis '{axis_name}'-axis "
+                        f"of a '{domain_type.value}' domain must contain a single value."
+                    )
+                if isinstance(axis, CompactAxis) and axis.num != 1:
                     raise ValueError(
-                        f"The 'values' field of the '{axis_name}'-axis "
+                        f"The 'values' field of the CompactAxis '{axis_name}'-axis "
                         f"of a '{domain_type.value}' domain must contain a single value."
```

Comment:
This should refer to `num` and not `values`, correct?

Reply:
:+1: